packages/verified-fetch/src/index.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * @packageDocumentation
+ *
+ * Exports a `createVerifiedFetch` function that returns a `fetch()` like API method {@link Helia} for fetching IPFS content.
+ *
+ * You may use any supported resource argument to fetch content:
+ *
+ * - CID instance
+ * - IPFS URL
+ * - IPNS URL
+ *
+ * @example Using a CID instance to fetch JSON
+ *
+ * ```typescript
+ * import { createVerifiedFetch } from '@helia/verified-fetch'
+ * import { CID } from 'multiformats/cid'
+ *
+ * const verifiedFetch = await createVerifiedFetch({
+ *  gateways: ['http://mygateway.info', 'http://trustless-gateway.link']
+ * })
+ *
+ * const cid = CID.parse('bafyFoo') // some image file
+ * const response = await verifiedFetch(cid)
+ * const json = await response.json()
```

Comment:
Do you think we should call the variable in the example `fetch` to really hammer the point home about API familiarity?

```suggestion
 * const fetch = await createVerifiedFetch({
 *  gateways: ['http://mygateway.info', 'http://trustless-gateway.link']
 * })
 *
 * const cid = CID.parse('bafyFoo') // some image file
 * const response = await fetch(cid)
 * const json = await response.json()
```

Reply:
I'm good with that, but we're shadowing the global fetch at this point which could cause confusion. Will wait for others' opinions before making this switch.